### PR TITLE
[google_sign_in_web] Add a no-op Android implementation

### PR DESCRIPTION
**Pull request by mklim**
## Related Issues

flutter/flutter#46898

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_sign_in/google_sign_in_web/android/src/main/java/io/flutter/plugins/google_sign_in_web/GoogleSignInWebPlugin.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import io.flutter.plugin.common.PluginRegistry.Registrar;
+
+/** GoogleSignInWebPlugin */
+public class GoogleSignInWebPlugin implements FlutterPlugin, MethodCallHandler {
+  @Override
+  public void onAttachedToEngine(FlutterPluginBinding flutterPluginBinding) {
+    final MethodChannel channel = new MethodChannel(flutterPluginBinding.getFlutterEngine().getDartExecutor(), "google_sign_in_web");
```
Comment (amirh) [inline packages/google_sign_in/google_sign_in_web/android/src/main/java/io/flutter/plugins/google_sign_in_web/GoogleSignInWebPlugin.java]: nit: delete the implementation

File: packages/google_sign_in/google_sign_in_web/android/src/main/java/io/flutter/plugins/google_sign_in_web/GoogleSignInWebPlugin.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import io.flutter.plugin.common.PluginRegistry.Registrar;
+
+/** GoogleSignInWebPlugin */
+public class GoogleSignInWebPlugin implements FlutterPlugin, MethodCallHandler {
```
Comment (amirh) [inline packages/google_sign_in/google_sign_in_web/android/src/main/java/io/flutter/plugins/google_sign_in_web/GoogleSignInWebPlugin.java]: nit: this doesn't need to extend MethodCallHandler (delete onMethodCall)

Comment (mklim) [reply packages/google_sign_in/google_sign_in_web/android/src/main/java/io/flutter/plugins/google_sign_in_web/GoogleSignInWebPlugin.java]: Done.

File: packages/google_sign_in/google_sign_in_web/android/src/main/java/io/flutter/plugins/google_sign_in_web/GoogleSignInWebPlugin.java (hunk 1 above)
Comment (mklim) [reply packages/google_sign_in/google_sign_in_web/android/src/main/java/io/flutter/plugins/google_sign_in_web/GoogleSignInWebPlugin.java]: Done.

**Comment by bsr203** [comment]
@mklim since you touched on google sigin package, can you please have a look at https://github.com/flutter/flutter/issues/41296 It is quite unusable without it, and hope it is a quick change for you. There was a pr about to merge and hope that help to. thanks.

**Pull request merged**